### PR TITLE
[IMP] selection input: dynamic highlight followup

### PR DESCRIPTION
**Pull request by yunchen4**
## Description:

This commit contains several follow-up work for the dynamic highlight task ([2884221](https://www.odoo.com/web#id=2884221&action=333&active_id=2328&model=project.task&view_type=form&cids=1&menu_id=4720)):
- Change range id type from UID to number
- Make hitting enter key act the same as clicking confirm button
- Change confirm button clicking cuntion name from `disable` to `confirm`
- Range id will start from 0 rather than 1
- Add tests for detecting duplicate range ids and hitting enter key

Odoo task ID : [3203266](https://www.odoo.com/web#id=3203266&action=333&active_id=2328&model=project.task&view_type=form&cids=1&menu_id=4720)

## review checklist

- [ ] feature is organized in plugin, or UI components
- [ ] support of duplicate sheet (deep copy)
- [ ] in model/core: ranges are Range object, and can be adapted (adaptRanges)
- [ ] in model/UI: ranges are strings (to show the user)
- [ ] undo-able commands (uses this.history.update)
- [ ] multiuser-able commands (has inverse commands and transformations where needed)
- [ ] new/updated/removed commands are documented
- [ ] exportable in excel
- [ ] translations (\_lt("qmsdf %s", abc))
- [ ] unit tested
- [ ] clean commented code
- [ ] track breaking changes
- [ ] doc is rebuild (npm run doc)
- [ ] status is correct in Odoo

### CODE REVIEW

File: src/components/selection_input/selection_input.ts [hunk 1]
```diff
@@ -108,9 +108,9 @@ export class SelectionInput extends Component<Props, SpreadsheetChildEnv> {
     const ranges = existingSelectionRange.length
       ? existingSelectionRange
       : this.props.ranges
-      ? this.props.ranges.map((xc, i) => ({
+      ? this.props.ranges.map((xc, id) => ({
           xc,
-          id: i.toString(),
+          id,
```
Comment (rrahir) [inline src/components/selection_input/selection_input.ts:113]: @LucasLefevre wdyt of changing the id to number ?

File: src/plugins/ui_feature/selection_input.ts [hunk 2]
```diff
@@ -182,13 +182,13 @@ export class SelectionInputPlugin extends UIPlugin implements StreamCallbacks<Se
    * Insert new inputs after the given index.
    */
   private insertNewRange(index: number, values: string[]) {
-    const currentMaxId = Math.max(0, ...this.ranges.map((range) => Number(range.id)));
+    const currentMaxId = Math.max(-1, ...this.ranges.map((range) => Number(range.id)));
```
Comment (rrahir) [inline src/plugins/ui_feature/selection_input.ts:185]: The fact that id started with 1 instaed of 0 was not an issue, the problem was that when you changed the behaviour, you added a +1 to the counter of colors, and I do'nt ythink it was necessary.
https://github.com/odoo/o-spreadsheet/pull/2066/files#diff-b0adb6ac15e927159a8739ac1456275e9371da9b7397a6c87b363fad6db1feb5R192

**Review by rrahir — COMMENTED** [review]
:+1:

File: src/components/selection_input/selection_input.ts (hunk 1 above)
Comment (LucasLefevre) [reply src/components/selection_input/selection_input.ts:113]: LGTM :+1:

**Comment by hokolomopo** [comment]
Typo in commit message `clicking cuntion name`

**Comment by rrahir** [comment]
@robodoo r+
